test/blackbox-tests/test-cases/runtest-cmd.t

```diff
@@ -0,0 +1,59 @@
+Here we test the features of the `dune runtest` command. 
+
+  $ cat > dune-project <<EOF
+  > (lang dune 3.16)
+  > EOF
+
+  $ cat > mytest.t <<EOF
+  >   $ echo "Hello, world!"
+  > "Goodbye, world!"
+  > EOF
+  $ mkdir -p tests/myothertest.t
+  $ cat > tests/myothertest.t/run.t <<EOF
+  >   $ echo "Hello, world!"
+  > "Goodbye, world!"
+  > EOF
+  $ cat > tests/filetest.t <<EOF
+  >   $ echo "Hello, world!"
+  > "Goodbye, world!"
+  > EOF
+
+Passing no arguments to `dune runtest` should be equivalent to `dune build
+@runtest`.
+
+  $ dune test 2>&1 | grep "^File"
+  File "mytest.t", line 1, characters 0-0:
+  File "tests/filetest.t", line 1, characters 0-0:
+  File "tests/myothertest.t/run.t", line 1, characters 0-0:
+
+Passing the name of a test should only run that test.
+
+  $ dune test mytest 2>&1 | grep "^File"
+  File "mytest.t", line 1, characters 0-0:
+  $ dune test mytest.t 2>&1 | grep "^File"
+  File "mytest.t", line 1, characters 0-0:
+  $ dune test tests/myothertest 2>&1 | grep "^File"
+  File "tests/myothertest.t/run.t", line 1, characters 0-0:
+  $ dune test tests/myothertest.t 2>&1 | grep "^File"
+  File "tests/myothertest.t/run.t", line 1, characters 0-0:
+
+We don't have to fully qualify tests in subdirectories.
```

Comment:
This should be forbidden. It's too confusing. We should just accept a valid source file or directory.

Reply:
This is now forbidden as the valid source file is checked for.